subgraphs/aave-governance/protocols/dydx-governance/src/dydx-governor.ts

```diff
@@ -0,0 +1,142 @@
+import { GovernanceStrategy } from "../../../generated/DydxGovernor/GovernanceStrategy";
+import { GovernanceFramework } from "../../../generated/schema";
+
+export function handleProposalCanceled(event: ProposalCanceled): void {
+  _handleProposalCanceled(event.params.id.toString(), event);
+}
+
+export function handleProposalCreated(event: ProposalCreated): void {
+  let executor = event.params.executor;
+  let quorumVotes = getQuorumFromContract(
+    event.address,
+    executor,
+    event.block.number.minus(BIGINT_ONE)
+  );
+
+  _handleProposalCreated(
+    event.params.id.toString(),
+    event.params.creator.toHexString(),
+    executor.toHexString(),
+    event.params.targets,
+    event.params.values,
+    event.params.signatures,
+    event.params.calldatas,
+    event.params.startBlock,
+    event.params.endBlock,
+    event.params.ipfsHash,
+    quorumVotes,
+    event
+  );
+}
+
+export function handleProposalExecuted(event: ProposalExecuted): void {
+  _handleProposalExecuted(event.params.id.toString(), event);
+}
+
+export function handleProposalQueued(event: ProposalQueued): void {
+  _handleProposalQueued(event.params.id.toString(), event.params.executionTime);
+}
+
+export function handleVoteEmitted(event: VoteEmitted): void {
+  let proposal = getOrCreateProposal(event.params.id.toString());
+
+  // if state is pending (i.e. the first vote), set state, quorum, delegates and tokenholders
+  if (proposal.state == ProposalState.PENDING) {
+    proposal.state = ProposalState.ACTIVE;
+    // Set snapshot for quorum, tokenholders and delegates
+    proposal.quorumVotes = getQuorumFromContract(
+      event.address,
+      Address.fromString(proposal.executor),
+      event.block.number.minus(BIGINT_ONE)
+    );
+    let governance = getGovernance();
+    proposal.tokenHoldersAtStart = governance.currentTokenHolders;
+    proposal.delegatesAtStart = governance.currentDelegates;
+  }
+
+  // Proposal will be updated as part of handler
+  _handleVoteEmitted(
+    proposal,
+    event.params.voter.toHexString(),
+    event.params.votingPower,
+    event.params.support,
+    event
+  );
+}
+
+// VotingDelayChanged (newVotingDelay, initiatorChange)
+export function handleVotingDelayChanged(event: VotingDelayChanged): void {
+  let governanceFramework = getGovernanceFramework(event.address.toHexString());
+  governanceFramework.votingDelay = event.params.newVotingDelay;
+  governanceFramework.save();
+}
+
+// Helper function that imports and binds the contract
+function getGovernanceFramework(contractAddress: string): GovernanceFramework {
+  let governanceFramework = GovernanceFramework.load(contractAddress);
+
+  if (!governanceFramework) {
+    governanceFramework = new GovernanceFramework(contractAddress);
+    let contract = DydxGovernor.bind(Address.fromString(contractAddress));
+    governanceFramework.name = contract.EIP712_DOMAIN_NAME();
+    governanceFramework.type = GOVERNANCE_NAME;
+    governanceFramework.version = "1";
```

Comment:
Does Aave governance have a concept of version?

Reply:
I don't see it in the contract which may make "1" a little misleading

"N/A" would probably be a little more accurate. Alternatively, could make version optional but that would be breaking change against OZ schema

Reply:
It doesnt use versions. I agree `N/A` is good default for unused entity fields